src/mlpack/core/optimizers/swats/swats_update.hpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file swats_update.hpp
+ * @author Marcus Edel
+ *
+ * SWATS update rule for Switches from Adam to SGD.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SWATS_SWATS_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_SWATS_SWATS_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * SWATS is a simple strategy which switches from Adam to SGD when a triggering
+ * condition is satisfied. The condition relates to the projection of Adam steps
+ * on the gradient subspace.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Keskar2017,
+ *   author  = {Nitish Shirish Keskar and Richard Socher},
+ *   title   = {Improving Generalization Performance by Switching from Adam to
+ *              {SGD}},
```

Comment:
Are the extra parentheses really required here?

Reply:
The ``{{`` will prevent BibTeX from adjusting the capitalization, so instead of ``Sgd`` we get ``SGD``.